keanu-project/src/main/java/io/improbable/keanu/util/io/ProtobufLoader.java

```diff
@@ -236,12 +245,16 @@ private Object getParameter(Parameter methodParameter,
 
         if ((paramAnnotation = methodParameter.getAnnotation(LoadVertexParam.class)) != null) {
             Object parameter = paramMap.get(paramAnnotation.value());
-            if (parameter == null) {
+            if (parameter == null && !paramAnnotation.isNullable()) {
                 throw new IllegalArgumentException("Failed to create vertex due to missing parent: "
                     + paramAnnotation.value());
-            } else {
-                return parameter;
             }
+            String annotationName = methodParameter.getAnnotation(LoadVertexParam.class).value();
+            if (annotationName.equals("label")) {
+                return VertexLabel.parseLabel((String) parameter);
```

Comment:
I'm definitely against special casing based on name - I'd much prefer we just leave this as a string and have the constructor call VertexLabel.parseLabel (which is a good function :-)